src/compiler/checker/lines_8000_12000.rs

```diff
@@ -333,16 +335,18 @@ impl TypeChecker {
             }
             .into();
             let type_rc: Rc<Type> = type_.into();
-            match &*type_rc {
-                Type::ObjectType(ObjectType::InterfaceType(InterfaceType::BaseInterfaceType(
-                    base_interface_type,
-                ))) => match &**base_interface_type.this_type.borrow_mut().as_ref().unwrap() {
-                    Type::TypeParameter(type_parameter) => {
-                        *type_parameter.constraint.borrow_mut() = Some(Rc::downgrade(&type_rc));
-                    }
-                    _ => panic!("Expected TypeParameter"),
-                },
-                _ => panic!("Expected BaseInterfaceType"),
+            if need_to_set_constraint {
```

Comment:
This is where the bug was that was causing the previously working type-checking of non-generic interface types to stop working, this setting of the `constraint` that had to happen after the the type was wrapped in an `Rc<Type>` (because it needed to get set to a weak reference to that `Rc<Type>`) should only have been happening in the "generic" case (corresponding to the `if` branch above)